src/core/domain-services/crux-gateway.ts

```diff
@@ -1,3 +1,4 @@
+import {decodeToken} from "blockstack";
```

Comment:
you shouldnt be depending on blockstack's library. Use `TokenSigner` lib directly. Check how its used in BasicKeyManager

Reply:
Token signer has been added in next PR